src/modules/leaderboard.ts

```diff
@@ -0,0 +1,182 @@
+import * as Discord from "discord.js";
+import * as Builders from "@discordjs/builders";
+
+import { CommandDescriptor } from "../bot.d";
+
+import * as utils from "./utils";
+
+const MAX_PEOPLE = 50;
+
+type UsersCharacterCount = Discord.Collection<Discord.Snowflake, number>;
+
+export async function getUsersCharacterCount(
+	guild: Discord.Guild,
+	onlyCountAfter?: Date
+): Promise<[UsersCharacterCount, number, number, Discord.Channel[]]> {
+	// TODO: checkpoint
+
+	const chars: UsersCharacterCount = new Discord.Collection();
+	const failed: (Discord.TextChannel | Discord.ThreadChannel)[] = [];
+	let msgCount = 0;
+
+	const channels = Array.from(
+		(await guild.channels.fetch()).filter(
+			(channel) => channel.isText() || channel.isThread()
+		)
+	).map((o) => o[1]) as (Discord.TextChannel | Discord.ThreadChannel)[];
+	const promises = channels.map(async (channel) => {
+		const messages = await channel.messages.fetch();
```

Comment:
How many messages would this fetch? I doubt this fetches all the messages since the beginning of the channel
In the [Discord docs](https://discord.com/developers/docs/resources/channel#get-channel-messages) it says the default of this endpoint is 50 messages while the maximum is 100.
For this to fetch all messages, it'd need to keep fetching until less than 50 messages are returned